sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/util/CertificateUtil.java

```diff
@@ -62,15 +61,14 @@ public static X509Certificate publicKeyFromPem(byte[] pem) {
         Pattern pattern = Pattern.compile("(?s)-----BEGIN CERTIFICATE-----.*-----END CERTIFICATE-----");
         Matcher matcher = pattern.matcher(new String(pem, StandardCharsets.UTF_8));
         if (!matcher.find()) {
-            LOGGER.logAndThrow(new IllegalArgumentException("PEM certificate provided does not contain -----BEGIN CERTIFICATE-----END CERTIFICATE----- block"));
+            throw LOGGER.logErrorAndThrow(new IllegalArgumentException("PEM certificate provided does not contain -----BEGIN CERTIFICATE-----END CERTIFICATE----- block"));
```

Comment:
logErrorAndThrow() will throw RuntimeError internally. A ClientLogger can't be static and the naming of ClientLogger variable has to be 'logger'.   It applies to other changes as well.

Reply:
I've changed it to return the RuntimeException to be thrown as part of this change, having it throw inside the method and needing to return null afterwards in a lot of cases isn't a good pattern and may result in strange code to be used in cases where final variables need to be set.